proxy/src/control/destination/background.rs

```diff
@@ -555,12 +560,27 @@ fn pb_to_addr_meta(
         .collect::<Vec<_>>();
     labels.sort_by(|(k0, _), (k1, _)| k0.cmp(k1));
 
-    let meta = Metadata {
-        metric_labels: DstLabels::new(labels.into_iter()),
-    };
+    let tls = pb.tls_identity.and_then(TlsIdentity::from_pb);
+
+    let meta = Metadata::new(DstLabels::new(labels.into_iter()), tls);
     Some((addr, meta))
 }
 
+impl TlsIdentity {
+    pub fn from_pb(pb: TlsIdentityPb) -> Option<Self> {
+        use conduit_proxy_controller_grpc::destination::tls_identity::Strategy;
+        pb.strategy.map(|strategy| match strategy {
+            Strategy::K8sPodNamespace(s) =>
+                TlsIdentity::K8sPodNamespace {
+                    controller_ns: Arc::from(s.controller_ns),
+                    pod_ns: Arc::from(s.pod_ns),
+                    pod_name: Arc::from(s.pod_name),
```

Comment:
Why are these `Arc`s?

Reply:
`Metadata` is cloned a lot, and I thought it was better to reference-count these than cloning the strings. However, I think it would (at least currently) be sufficient to place one `Arc` around the entire `TlsIdentity`, will change that.

Reply:
> Metadata is cloned a lot, and I thought it was better to reference-count these than cloning the strings.

I don't have a particular opinion on this specific case (yet), but this sounds unlikely to me.  I can imagine `Metadata` being cloned throughout service discovery, but this shouldn't be "a lot" -- as in not per-request. We should be somewhat concerned about per-request allocations, but I think it may be premature to be overly conservative with things that are generally scoped at the balancer or connection.